src/garage/tf/optimizers/conjugate_gradient_optimizer.py

```diff
@@ -133,6 +133,16 @@ def eval(x):
 
         return eval
 
+    def __getstate__(self):
+        """Object.__getstate__."""
+        new_dict = self.__dict__.copy()
+        del new_dict['opt_fun']
+        return new_dict
+
+    def __setstate__(self, state):
```

Comment:
Do you need to re-create `opt_fun` here?
If you *don't*, then you should be able to delete this `__setstate__` implementation, since this is the "default."
Note that you'll still have a `__getstate__` implementation then, despite not having a `__setstate__` implementation.

Reply:
Good catch, will remove `__getstate__` since `opt_fun` is recreated somewhere else.